File: API/DTOs/WateringDto.cs
using API.Models;
using System.ComponentModel.DataAnnotations;

namespace API.DTOs
{
    // Watering Schedules
    public class WateringScheduleResponseDto
    {
        public int ScheduleId { get; set; }
        public int PlantId { get; set; }
        public TimeSpan StartTime { get; set; }
        public int IntervalHours { get; set; }
        public int AmountMl { get; set; }
        public bool Enabled { get; set; }
        public int[] DaysOfWeek { get; set; } = Array.Empty<int>();
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }

    public class CreateWateringScheduleDto
    {
        [Required]
        public int PlantId { get; set; }

        [Required]
        public TimeSpan StartTime { get; set; }

        public int IntervalHours { get; set; }
        public int AmountMl { get; set; }
        public bool Enabled { get; set; } = true;
        public int[] DaysOfWeek { get; set; } = Array.Empty<int>();
    }

    public class UpdateWateringScheduleDto
    {
        [Required]
        public TimeSpan StartTime { get; set; }

        public int IntervalHours { get; set; }
        public int AmountMl { get; set; }
        public bool Enabled { get; set; }
        public int[] DaysOfWeek { get; set; } = Array.Empty<int>();
    }

    // Watering Events
    public class WateringEventResponseDto
    {
        public int EventId { get; set; }
        public int PlantId { get; set; }
        public int? ScheduleId { get; set; }
        public DateTime Timestamp { get; set; }
        public int AmountMl { get; set; }
        public WateringMode Mode { get; set; }
        public int? DurationSeconds { get; set; }
        public WateringStatus Status { get; set; }
        public string? Notes { get; set; }
    }

    public class CreateWateringEventDto
    {
        [Required]
        public int PlantId { get; set; }

        public int? ScheduleId { get; set; }
        public int AmountMl { get; set; }

        [Required]
        public WateringMode Mode { get; set; } = WateringMode.Automatic;

        public int? DurationSeconds { get; set; }
        public WateringStatus Status { get; set; } = WateringStatus.Completed;
        public string? Notes { get; set; }
    }
}

File: Mobile/composeApp/src/androidMain/kotlin/com/plantiq/data/model/WateringDto.kt
package com.plantiq.data.model

import com.google.gson.annotations.SerializedName

data class WateringScheduleResponseDto(
    @SerializedName("scheduleId") val scheduleId: Int,
    @SerializedName("plantId") val plantId: Int,
    @SerializedName("startTime") val startTime: String,
    @SerializedName("intervalHours") val intervalHours: Int,
    @SerializedName("amountMl") val amountMl: Int,
    @SerializedName("enabled") val enabled: Boolean,
    @SerializedName("daysOfWeek") val daysOfWeek: List<Int>,
    @SerializedName("repeatCount") val repeatCount: Int,
    @SerializedName("createdAt") val createdAt: String,
    @SerializedName("updatedAt") val updatedAt: String
)

data class CreateWateringScheduleDto(
    @SerializedName("plantId") val plantId: Int,
    @SerializedName("startTime") val startTime: String,
    @SerializedName("intervalHours") val intervalHours: Int,
    @SerializedName("amountMl") val amountMl: Int,
    @SerializedName("enabled") val enabled: Boolean = true,
    @SerializedName("daysOfWeek") val daysOfWeek: List<Int> = emptyList(),
    @SerializedName("repeatCount") val repeatCount: Int = 1
)

data class UpdateWateringScheduleDto(
    @SerializedName("startTime") val startTime: String,
    @SerializedName("intervalHours") val intervalHours: Int,
    @SerializedName("amountMl") val amountMl: Int,
    @SerializedName("enabled") val enabled: Boolean,
    @SerializedName("daysOfWeek") val daysOfWeek: List<Int>,
    @SerializedName("repeatCount") val repeatCount: Int
)

data class WateringEventResponseDto(
    @SerializedName("eventId") val eventId: Int,
    @SerializedName("plantId") val plantId: Int,
    @SerializedName("timestamp") val timestamp: String,
    @SerializedName("amountMl") val amountMl: Int,
    @SerializedName("mode") val mode: String,
    @SerializedName("status") val status: String
)
